specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2019-05-06-preview/searchindex.json

```diff
@@ -930,6 +996,15 @@
           "readOnly": true,
           "x-ms-client-name": "Highlights",
           "description": "Text fragments from the document that indicate the matching search terms, organized by each applicable field; null if hit highlighting was not enabled for the query."
+        },
+        "@search.features": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/SearchFeatures"
+          },
+          "readOnly": true,
```

Comment:
I don't really get how this is supposed to work.  We're going to have a `Features` property with a single member called `additionalProperties` of type `SearchFeatures`?

Reply:
I wasn't too sure how to model this. After asking around in my team, I was under the impression the additionalProperties field was used to define a dictionary. We can see what the @search.features look like in the JSON payload here:
https://github.com/Azure/azure-rest-api-specs/pull/8793/files#diff-e26d55dedb47c9d72513f630d461dd22R93-R104
It's basically a dictionary of dictionaries, where the first level keys are field names (which is dynamic since it's based on the customer's index definition), and the second level keys are the type of features (which is static). I might have missed the mark on how to model that.

Reply:
I see - thanks for explaining.  That looks good.